planner: add INL_MERGE_JOIN deprecation warning to optimizer hints (#23421)
pingcap/docs: optimizer-hints.md

### INL_MERGE_JOIN

> **Warning:** 
>

> > Starting from TiDB v8.3.0, the `INL_MERGE_JOIN` hint is deprecated because it might return incorrect results. Do not use this hint in new queries. It might be removed in a future release.

Reviewer: ```suggestion
> Starting from TiDB v8.3.0, the `INL_MERGE_JOIN` hint is deprecated and no longer takes effect, because it might return incorrect results. If a query specifies this hint, TiDB ignores it and selects another join algorithm.
```

> The `INL_MERGE_JOIN(t1_name [, tl_name])` hint tells the optimizer to use the index nested loop merge join algorithm. The conditions for using this algorithm are the same with the conditions for using the index nested loop join algorithm.

Reviewer: ```suggestion
Before v8.3.0, the `INL_MERGE_JOIN(t1_name [, tl_name])` hint tells the optimizer to use the index nested loop merge join algorithm. The conditions for using this algorithm are the same with the conditions for using the index nested loop join algorithm.
```

### INL_MERGE_JOIN

> > **Warning:** The `INL_MERGE_JOIN` hint has been deprecated since TiDB v8.3.0 because it might return incorrect results. Do not use this hint in new queries. It might be removed in a future release.

Reviewer: ```suggestion
> **Warning:** 
>
> Starting from TiDB v8.3.0, the `INL_MERGE_JOIN` hint is deprecated because it might return incorrect results. Do not use this hint in new queries. It might be removed in a future release.
```
